odkx-src/injected-interfaces.rst

```diff
@@ -733,3 +786,520 @@ removeFirstQueuedAction
+
+This function is used to retrieve the roles associated with the current user.
+It takes in success and failure callback functions to handle the result of the role retrieval.
+The function internally uses the ``getOdkDataIf().getRoles`` method to perform the retrieval and passes the request to the ``queueRequest`` function for handling callbacks.
+
+getDefaultGroup
+~~~~~~~~~~~~~~~~
+
+  **Parameters**:
+
+  - successCallbackFn: A success callback function that is called when the default group is successfully retrieved.
+  - failureCallbackFn: A failure callback function that is called when there is an error in retrieving the default group.
+
+This function is used to retrieve the default group for the current user.
+It takes in success and failure callback functions to handle the result of the default group retrieval.
+The function internally uses the **getOdkDataIf().getDefaultGroup** method to perform the retrieval and passes the request to the **queueRequest** function for handling callbacks.
+
+getUsers
+~~~~~~~~
+
+  **Parameters**:
+
+  - successCallbackFn: A success callback function that is called when the list of users is successfully retrieved.
+  - failureCallbackFn: A failure callback function that is called when there is an error in retrieving the list of users.
+
+This function is used to retrieve the list of users.
+It takes in success and failure callback functions to handle the result of the user list retrieval.
+The function internally uses the ``getOdkDataIf().getUsers`` method to perform the retrieval and passes the request to the ``queueRequest`` function for handling callbacks.
+
+getAllTableIds
+~~~~~~~~~~~~~~~
+
+  **Parameters**:
+
+  - successCallbackFn: A success callback function that is called when the list of table IDs is successfully retrieved.
+  - failureCallbackFn: A failure callback function that is called when there is an error in retrieving the list of table IDs.
+
+This function is used to retrieve the list of all available table IDs.
+It takes in success and failure callback functions to handle the result of the table ID retrieval.
+The function internally uses the ``getOdkDataIf().getAllTableIds`` method to perform the retrieval and passes the request to the ``queueRequest`` function for handling callbacks.
+
+query
+~~~~~~
+
+  **Parameters**:
+
+  - tableId: The ID of the table on which to perform the query.
+  - whereClause: The SQL WHERE clause for filtering the data.
+  - sqlBindParams: An array of SQL bind parameters to be used in the query.
+  - groupBy: The SQL GROUP BY clause for grouping the data.
+  - having: The SQL HAVING clause for filtering grouped data.
+  - orderByElementKey: The key for ordering the data.
+  - orderByDirection: The direction (ASC or DESC) for ordering the data.
+  - limit: The maximum number of rows to return.
+  - offset: The starting offset of rows to return.
```

Comment:
defining offset as offset is a bit circular, maybe explain instead as "the starting rows to return" or "how many rows to initially skip" or something like that that describes offset.